frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FormCreator.tsx

```diff
@@ -52,23 +72,67 @@ export function FormCreator({
     });
   };
 
+  const handleDragEnd: OnDragEndResponder = ({ source, destination }) => {
+    const oldOrder = source.index;
+    const newOrder = destination?.index ?? source.index;
+
+    const reorderedFields = reorderFields(
+      formSettings.fields,
+      oldOrder,
+      newOrder,
+    );
+    setFormSettings({
+      ...formSettings,
+      fields: reorderedFields,
+    });
+  };
+
+  const sortedParams = useMemo(
+    () => params.sort(sortActionParams(formSettings)),
+    [params, formSettings],
+  );
+
   return (
     <FormCreatorWrapper>
-      {params.map(param => (
-        <FormItem
-          key={param.id}
-          param={param}
-          fieldSettings={
-            formSettings.fields?.[param.id] ?? getDefaultFieldSettings()
-          }
-          onChange={(newSettings: FieldSettings) =>
-            handleChangeFieldSettings(param.id, newSettings)
-          }
-        />
-      ))}
-      {!params.length && (
-        <EmptyFormPlaceholder onExampleClick={onExampleClick} />
-      )}
+      <DragDropContext onDragEnd={handleDragEnd}>
+        <Droppable droppableId="action-form-droppable">
```

Comment:
copied this structure right from `ChartSettingFieldsPicker.jsx` and simply added type definitions